interfaces/devdoc/string_utils.md

```diff
@@ -0,0 +1,61 @@
+`string_utils` requirements
+============
+
+## Overview
+
+`string_utils` is a collection of string utils. The set of requirements in this document is far from complete.
+
+## Exposed API
+
+```c
+/*produces a string as if printed by printf*/
+char* sprintf_char_function(const char* format, ...);
+
+/*produces a string as if printed by printf (will also verify arguments)*/
+#define sprintf_char(format, ...) (0?printf((format), ## __VA_ARGS__):0, sprintf_char_function((format), ##__VA_ARGS__))
+
+/*produces a string as if printed by wprintf*/
+wchar_t* sprintf_wchar_function(const wchar_t* format, ...);
+#define sprintf_wchar(format, ...) (0?wprintf((format), ##__VA_ARGS__):0, sprintf_wchar_function((format), ##__VA_ARGS__))
+
+MOCKABLE_INTERFACE(string_utils_printf,
+    /*produces a string as if printed by vprintf*/
+    FUNCTION(, char*, vsprintf_char, const char*, format, va_list, va),
+    /*produces a string as if printed by vwprintf*/
+    FUNCTION(, wchar_t*, vsprintf_wchar, const wchar_t*, format, va_list, va)
+)
+
+/*takes a FILETIME, returns a nice string representation of it*/
+MOCKABLE_FUNCTION(, char*, FILETIME_toAsciiArray, const FILETIME*, fileTime);
+
+/*takes a FILETIME, returns a string in UTC time zone*/
+MOCKABLE_FUNCTION(, char*, FILETIME_to_string_UTC, const FILETIME*, fileTime);
+
+MOCKABLE_INTERFACE(string_utils_convert,
+    /*produces the wchar_t* string representation of source (which is assumed to be multibyte). Returns NULL on any failure.*/
+    FUNCTION(, wchar_t*, mbs_to_wcs, const char*, source),
+
+    /*produces the multibyte char* string representation of source. Returns NULL on any failure.*/
+    FUNCTION(, char*, wcs_to_mbs, const wchar_t*, source)
+)
+```
+
+### FILETIME_to_string_UTC
+```c
+MOCKABLE_FUNCTION(, char*, FILETIME_to_string_UTC, const FILETIME*, fileTime);
+```
+
+`FILETIME_to_string_UTC` returns a human readable representation of `fileTime`. Here's an example: `(SYSTEMTIME){ /* 2022-06-09T18:59:48.476Z */ .wYear = 2022, .wMonth = 06, .wDayOfWeek = 4, .wDay = 09, .wHour = 18, .wMinute = 59, .wSecond = 48, .wMilliseconds = 476 }`. The returned string needs to be `free`d by the caller.
```

Comment:
> human readable representation [](http://example.com/codeflow?start=34&length=30)

According to your comment in the other file, it's also a C initializer format, which is maybe interesting to call out here.

Reply:
Removed the designated initializer part.